python/pyspark/sql/connect/dataframe.py

```diff
@@ -468,7 +468,7 @@ def take(self, num: int) -> List[Row]:
     def join(
         self,
         other: "DataFrame",
-        on: Optional[Union[str, List[str], Column]] = None,
+        on: Optional[Union[str, List[str], Column, List[Column]]] = None,
```

Comment:
Just dropping a comment. It should actually be able to accept a tuple too .. or `Iterable` .. but I am fine because that's what's already done in PySpark DataFrame.

Reply:
Given this matches what PySpark does I will leave it as it now.